extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -329,6 +331,12 @@ public void configurationDescriptorBuilding(
                     .filter(i -> i.isDefault())
                     .findFirst();
             collectDialectConfigForPersistenceXml(puName, xmlDescriptor);
+            Optional<FormatMapperKind> jsonMapper = jsonMapperKind(capabilities);
+            Optional<FormatMapperKind> xmlMapper = xmlMapperKind(capabilities);
+            jsonMapper.flatMap(FormatMapperKind::requiredBeanType)
+                    .ifPresent(type -> unremovableBeans.produce(UnremovableBeanBuildItem.beanClassNames(type)));
+            xmlMapper.flatMap(FormatMapperKind::requiredBeanType)
+                    .ifPresent(type -> unremovableBeans.produce(UnremovableBeanBuildItem.beanClassNames(type)));
```

Comment:
Can't say I'm a big fan of this pattern as I think it makes a relatively simple thing harder.
Things would have been a bit simpler if you returned directly the bean type from the methods IMHO.

But I suppose it's a matter of taste so I can live with it :)